### PR TITLE
Moved common package for bots, added chat id into db

### PR DESCRIPTION
**Pull request by esuwu**

### CODE REVIEW

**Comment by esuwu** [comment]
I'd also like to add `/help` command before merging

File: pkg/messaging/client.go [hunk 1]
```diff
@@ -41,17 +36,7 @@ func StartMessagingClient(ctx context.Context, nanomsgURL string, bot *telebot.B
...
-				if !botEnv.ReceivedChat {
-					log.Println("haven't received a chat id yet")
-					continue
-				}
-
-				_, err = bot.Send(botEnv.Chat, string(msg))
-				if err != nil {
-					log.Printf("failed to send a message to telegram, %v", err)
-				}
+				bots.TgBotEnvironment.SendMessageTg(msg)
```
Comment (alexeykiselev) [inline pkg/messaging/client.go:39]: `Bots` is so telegram specific. Better make `Bots` a slice of `Bot` interfaces with `SendMessage` method and call it here for every bot in collection.

File: pkg/messaging/bots.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"log"
+
+	"gopkg.in/telebot.v3"
+	"nodemon/pkg/entities"
+	"nodemon/pkg/storing/chats"
+)
+
+type Bots struct {
```
Comment (alexeykiselev) [inline pkg/messaging/bots.go:11]: Convert to the slice of `Bot` interfaces.

File: pkg/messaging/bots.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	TgBotEnvironment *TelegramBotEnvironment
+	// here will be the discord Bot
+}
+
+func NewBots(tgBotEnvironment *TelegramBotEnvironment) *Bots {
+	return &Bots{TgBotEnvironment: tgBotEnvironment}
+}
+
+type TelegramBotEnvironment struct {
```
Comment (alexeykiselev) [inline pkg/messaging/bots.go:20]: This package is not a good place for telegram specific structures.

File: pkg/storing/chats/chats.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+// AfterFind required by Hare function
+func (c *chat) AfterFind(_ *hare.Database) error {
+	return nil
+}
+
+type Storage struct {
+	db *hare.Database
+}
+
+func NewStorage(path string) (*Storage, error) {
```
Comment (alexeykiselev) [inline pkg/storing/chats/chats.go:38]: Is this storage usable by other than telegram bot? If not, have to be moved to `tg_bot/internal`

**Review by alexeykiselev — CHANGES_REQUESTED** [review]

File: pkg/storing/chats/chats.go (hunk 4 above)
Comment (esuwu) [reply pkg/storing/chats/chats.go:38]: It is not yet. But I was going to use one `chats` table for both bots. So this `storage` is going to be common for both

File: cmd/bots/bots.go [hunk 5]
```diff
@@ -3,15 +3,14 @@ package main
 import (
```
Comment (nickeskov) [inline cmd/bots/bots.go:3]: Why did you move this file? I think every app should have own directory. Create new directory for telegram bot.

File: cmd/bots/internal/tg_bot/bot.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	"gopkg.in/telebot.v3"
+	"log"
+	"nodemon/pkg/entities"
+	"nodemon/pkg/storing/chats"
+)
+
+type TelegramBotEnvironment struct {
+	ChatStorage *chats.Storage
+	Bot         *telebot.Bot
+	ShutUp      bool
```
Comment (nickeskov) [inline cmd/bots/internal/tg_bot/bot.go:13]: I think we should be a bit more polite. Rename this variable to a corresponding neutral word.

File: pkg/messaging/bots.go [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package messaging
+
+type Bots []Bot
+
+func NewBots(bots ...Bot) Bots {
+	return bots
+}
```
Comment (nickeskov) [inline pkg/messaging/bots.go:7]: I think this function is unnecessary. You can easily convert `[]Bot` slice to `Bot` type.

File: cmd/bots/bots.go [hunk 8]
```diff
@@ -33,45 +32,49 @@ func run() error {
...
 			log.Printf("failed to start messaging service: %v", err)
 			return
 		}
 	}()
 
-	log.Println("Telegram bot started")
-	bot.Start()
+	for _, bot := range bots {
+		bot.Start()
+	}
```
Comment (nickeskov) [inline cmd/bots/bots.go:77]: I think one app for all bot isn't good idea. It would be better if you create new app for each bot.

File: cmd/tg_bot/internal/handlers/handlers.go [hunk 9]
```diff
@@ -1,25 +1,59 @@
...
+			return c.Send("Hello! I remember this chat.")
+		}
+		chatID := entities.ChatID(c.Chat().ID)
+
+		err = environment.ChatStorage.InsertChatID(chatID, entities.TelegramPlatform)
+		if err != nil {
+			log.Printf("failed to insert chat id into db: %v", err)
+			return c.Send("I failed to save this chat id")
+		}
+		return c.Send("Hello! This new chat has been saved for alerting.")
```
Comment (nickeskov) [inline cmd/tg_bot/internal/handlers/handlers.go:29]: It looks very unsafe. Anyone can interact with bot and subscribe for alerts.

Comment (nickeskov) [inline cmd/tg_bot/internal/handlers/handlers.go:29]: It doesn't look safe way. Anyone can interact with bot and subscribe for alerts.

File: cmd/tg_bot/tg_bot.go [hunk 10]
```diff
@@ -33,45 +32,47 @@ func run() error {
...
 	flag.StringVar(&botToken, "bot-token", "", "Temporarily: the default token is the current token")
 	flag.StringVar(&publicURL, "public-url", "", "Default is https://mainnet-go-htz-fsn1-1.wavesnodes.com/bot")
+	flag.StringVar(&storagePath, "storage", "", "Path to storage")
 	flag.Parse()
 
-	if botToken == "" || publicURL == "" {
-		log.Println("Invalid bot token or public URL")
+	if botToken == "" {
+		log.Println("Invalid bot token")
+		return errors.New("invalid parameters")
```
Comment (nickeskov) [inline cmd/tg_bot/tg_bot.go:47]: Move this error to variable

Comment (nickeskov) [inline cmd/tg_bot/tg_bot.go:47]: Move this error to variable

File: pkg/storing/nodes/nodes.go [hunk 11]
```diff
@@ -2,6 +2,7 @@ package nodes
 
 import (
 	"log"
+	"nodemon/pkg/storing/common"
```
Comment (nickeskov) [inline pkg/storing/nodes/nodes.go:5]: Please, sort imports.

File: pkg/storing/chats/chats.go [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+			return nil, err
+		}
+		chats = append(chats, ch)
+	}
+	for _, chat := range chats {
+		if chat.Chat.Platform == platform {
+			return &chat.Chat.ChatID, nil
+		}
+	}
+	return nil, nil
```
Comment (nickeskov) [inline pkg/storing/chats/chats.go:84]: Please, create specific error to show that chatID doesn't exist.

Comment (nickeskov) [inline pkg/storing/chats/chats.go:84]: Please, create specific error to show that chatID doesn't exist.

**Review by nickeskov — CHANGES_REQUESTED** [review]
Also, fix imports order, please.

File: cmd/tg_bot/internal/handlers/handlers.go (hunk 9 above)
Comment (esuwu) [reply cmd/tg_bot/internal/handlers/handlers.go:29]: Yes, I am aware. Will be fixed in the next PR.

Comment (esuwu) [reply cmd/tg_bot/internal/handlers/handlers.go:29]: dublicated

File: cmd/tg_bot/internal/environment.go [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+	chatID, err := tgEnv.ChatStorage.FindChatID(entities.TelegramPlatform)
+	if err != nil && err != chats.ErrorChatNotFound {
+		log.Printf("failed to find chat id: %v", err)
+		return
+	}
+
+	if err == chats.ErrorChatNotFound {
+		log.Println("have not received a chat id yet")
+		return
+	}
```
Comment (nickeskov) [inline cmd/tg_bot/internal/environment.go:41]: I think it looks prettier.
```suggestion
	if err != nil {
		if errors.Is(err, chats.ErrorChatNotFound) {
			log.Println("have not received a chat id yet")
		} else {
			log.Printf("failed to find chat id: %v", err)
		}
		return
	}
```

File: cmd/tg_bot/internal/handlers/handlers.go (hunk 9 above)
Comment (nickeskov) [reply cmd/tg_bot/internal/handlers/handlers.go:29]: yep

**Review by nickeskov — CHANGES_REQUESTED** [review]

**Review by nickeskov — APPROVED** [review]
